fe1-web/model/network/method/message/data/MessageRegistry.ts

```diff
@@ -0,0 +1,35 @@
+import { ActionType, ObjectType } from './MessageData';
+import { ExtendedMessage } from '../ExtendedMessage';
+
+type HandleFunction = (msg: ExtendedMessage) => boolean;
+
+/**
+ * Represents an entry of the MessageRegistry.
+ */
+interface MessageEntry {
+  ingestion: HandleFunction; // Function to handle this type of message
+}
+
+// Generates a string key of the form "object, action" for the MessageRegistry mapping
+const k = (object: ObjectType, action: ActionType): string => `${object}, ${action}`;
+
+/**
+ * Registry of message data classes and their corresponding properties.
+ */
+export class MessageRegistry {
+  private readonly mapping: Map<string, MessageEntry> = new Map<string, MessageEntry>();
+
+  addHandler(object: ObjectType, action: ActionType, handleFunc: HandleFunction) {
+    this.mapping.set(k(object, action), { ingestion: handleFunc });
```

Comment:
as the `MessageEntry` interface grows, we may need to just update that specific field, i.e. do a `this.mapping.get` first and update the entry.

Reply:
That's true, I realized it while trying to add signatures. I'll change it now.